src/libuv/gen_build_yaml.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python2.7
+
+# Copyright 2020 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import sys
+import yaml
+import glob
+
+out = {}
```

Comment:
needing to maintain this file is going to be a pain (especially because it hand lists the source files and there are 3 different libraries for 3 different systems (linux, darwin, windows).

Reply:
how does the selection between the 3 flavors of the library happens?

Reply:
Yes, it is a pain to maintain the list of source files for different systems, but I don't find a better way to solve it. Even [the official libuv build files](https://github.com/libuv/libuv/blob/v1.35.0/CMakeLists.txt#L87-L112) list them manually.

On each system, only corresponding library will be built from the source, as their build rules are guarded behind [the system check](https://github.com/grpc/grpc/blob/19df07ef98532b109c1e8556a596239c5726bba7/templates/Makefile.template#L1565). The other two libraries (both sources and build rules) will be ignored in the build.

Reply:
ok, at least add a comment where are the definitions for different libraries in this file coming from? (is it from libuv's CMakeLists.txt?). In the future, folks will wonder what is the magic in this file based on.